dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -220,3 +232,111 @@ def __exit__(
 
     def should_load_data_to_staging_dataset(self, table: TTableSchema) -> bool:
         return False
+
+    #
+    # state stuff
+    #
+
+    def _write_to_json_file(self, filepath: str, data: DictStrAny) -> None:
+        dirname = os.path.dirname(filepath)
+        self.fs_client.makedirs(dirname, exist_ok=True)
+        self.fs_client.write_text(filepath, json.dumps(data), "utf-8")
+
+    def complete_load(self, load_id: str) -> None:
+        # store current state
+        self.store_current_state()
+
+        # write entry to load "table"
+        # TODO: this is also duplicate across all destinations. DRY this.
+        load_data = {
+            "load_id": load_id,
+            "schema_name": self.schema.name,
+            "status": 0,
+            "inserted_at": pendulum.now().isoformat(),
+            "schema_version_hash": self.schema.version_hash,
+        }
+        filepath = (
+            f"{self.dataset_path}/{self.schema.loads_table_name}/{self.schema.name}.{load_id}.jsonl"
+        )
+
+        self._write_to_json_file(filepath, load_data)
+
+    #
+    # state read/write
+    #
+
+    def _get_state_file_name(self, pipeline_name: str, hash: str) -> Tuple[str, str]:
+        """gets tuple of dir and fullpath for schema file for a given hash"""
+        safe_hash = "".join(
+            [c for c in hash if re.match(r"\w", c)]
+        )  # remove all special chars from hash
+        return (
+            f"{self.dataset_path}/{self.schema.state_table_name}/{pipeline_name}__{safe_hash}.jsonl"
+        )
+
+    def store_current_state(self) -> None:
+        # get state doc from current pipeline
```

Comment:
This is a bit of a hack to get the state from the current pipeline instance and store it as json. You can see that I send an empty load job for the state table. I am not sure we should do it this way but it saves us dealing with different output fileformats and paths.

Reply:
I have done this, but somehow this feels like a bit strange solution too. The place where this state resource is extracted is right before the load package gets committed for normalization, so why do we not just put the current pipeline state in the load package state right before committing in this place?

Reply:
is it beneficial to use zip for state files at all?